python/hail/utils/misc.py

```diff
@@ -126,26 +123,11 @@ def fmt_field(field):
 
 
 def get_nice_field_error(obj, item):
-    from hail.api2 import MatrixTable, GroupedMatrixTable, Table, GroupedTable
-    from hail.expr import TStruct
-    if isinstance(obj, MatrixTable):
-        class_name = 'MatrixTable'
-        index_obj = obj
-    elif isinstance(obj, GroupedMatrixTable):
-        class_name = 'GroupedMatrixTable'
-        index_obj = obj._parent
-    elif isinstance(obj, Table):
-        class_name = 'Table'
-        index_obj = obj
-    elif isinstance(obj, GroupedTable):
-        class_name = 'GroupedTable'
-        index_obj = obj._parent
-    else:
-        raise NotImplementedError(obj)
+    class_name, index_obj, _ = get_obj_metadata(obj)
 
-    fields = obj._fields.keys()
+    field_names = obj._fields.keys()
     dd = defaultdict(lambda: [])
-    for f in fields:
+    for f in field_names:
         dd[f.lower()].append(f)
 
     def fmt_field(field):
```

Comment:
Sorry, I should have been clearer. `fmt_field` is also duplicated. I still think the rest of the code is very similar, but it might be more confusing to try and combine the two methods.

Reply:
fmt_field does different things, though. I tried to extract it and it required the dataset and whether to print [row field] vs [row] and ended up looking super messy.

Reply:
ok.